synapse/storage/util/id_generators.py

```diff
@@ -671,14 +671,50 @@ def get_next_txn(self, txn: LoggingTransaction) -> int:
 
         return self._return_factor * next_id
 
-    def _mark_id_as_finished(self, next_id: int) -> None:
-        """The ID has finished being processed so we should advance the
+    def get_next_mult_txn(self, txn: LoggingTransaction, n: int) -> List[int]:
```

Comment:
I wonder if `get_next_txn` should call this instead of duplicating some of the logic? (Same kind of goes for the cache & stream functions above TBH)